src/resolve/resolved-varlink.c

```diff
@@ -1211,6 +1227,42 @@ static int verify_polkit(sd_varlink *link, sd_json_variant *parameters, const ch
                                 &m->polkit_registry);
 }
 
+static int vl_method_browse_services(sd_varlink* link, sd_json_variant* parameters, sd_varlink_method_flags_t flags, void* userdata) {
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "domain",  SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(LookupParametersBrowse, domain),  0 },
+                { "type",    SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(LookupParametersBrowse, type),    0 },
+                { "ifindex", _SD_JSON_VARIANT_TYPE_INVALID, sd_json_dispatch_int,          offsetof(LookupParametersBrowse, ifindex), 0 },
+                { "flags",   _SD_JSON_VARIANT_TYPE_INVALID, sd_json_dispatch_uint64,       offsetof(LookupParametersBrowse, flags),   0 },
+                {}
+        };
+
+        LookupParametersBrowse p = {};
+        Manager *m;
+        int r = 0;
+
+        assert(link);
+
+        /* if the client didn't set the more flag, it is using us incorrectly */
+        if (!FLAGS_SET(flags, SD_VARLINK_METHOD_MORE))
+                return sd_varlink_error(link, SD_VARLINK_ERROR_EXPECTED_MORE, NULL);
+
+        m = sd_varlink_server_get_userdata(sd_varlink_get_server(link));
+        ASSERT_PTR(m);
```

Comment:
ASSERT_PTR() has this nice feature that it returns what you pass in, hence you can use it like this:

```c
m = ASSERT_PTR(sd_varlink_server_get_userdata(sd_varlink_get_server(link)));
```

(or in other words: the way you used ASSERT_PTR() here is pointless, you could have used a simple assert() for that, after all ASSERT_PTR() exists because of the ret val only really)